src/transformers/modeling_tf_bart.py

```diff
@@ -0,0 +1,1215 @@
+    is a :class:`TFDecoderLayer`.
+    Args:
+        config: BartConfig
+        embed_tokens: output embedding
+    """
+
+    def __init__(self, config: BartConfig, embed_tokens, **kwargs):
+        super().__init__(**kwargs)
+        self.layerdrop = config.decoder_layerdrop
+        self.padding_idx = config.pad_token_id
+        self.max_target_positions = config.max_position_embeddings
+        self.embed_tokens = embed_tokens
+        self.embed_scale = math.sqrt(config.d_model) if config.scale_embedding else 1.0
+        self.embed_positions = TFLearnedPositionalEmbedding(
+            config.max_position_embeddings,
+            config.d_model,
+            self.padding_idx,
+            config.extra_pos_embeddings,
+            name="embed_positions",
+        )
+        self.layers = [TFDecoderLayer(config, name=f"layers.{i}") for i in range(config.decoder_layers)]
+        self.layernorm_embedding = (
+            tf.keras.layers.LayerNormalization(epsilon=1e-5, name="layernorm_embedding")
+            if config.normalize_embedding
+            else tf.identity
+        )
+        self.layer_norm = (
+            tf.keras.layers.LayerNormalization(epsilon=1e-5, name="layer_norm")
+            if config.add_final_layer_norm
+            else None
+        )
+
+        self.dropout = tf.keras.layers.Dropout(config.dropout)
+        self.output_hidden_states = config.output_hidden_states
+        self.output_attentions = config.output_attentions
+        self.use_cache = config.use_cache
+        self.do_blenderbot_90_layernorm = config.do_blenderbot_90_layernorm
+
+    def call(
+        self,
+        input_ids,
+        encoder_hidden_states,
+        encoder_padding_mask,
+        decoder_padding_mask,
+        decoder_causal_mask,
+        decoder_cached_states=None,
+        use_cache=False,
+        output_attentions=False,
+        output_hidden_states=False,
+        return_dict=None,
+        training=False,
+        **unused,
+    ):
+        output_attentions = output_attentions if output_attentions is not None else self.output_attentions
+        output_hidden_states = output_hidden_states if output_hidden_states is not None else self.output_hidden_states
+        use_cache = use_cache if use_cache is not None else self.use_cache
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+        if use_cache:
+            assert not training, "Training + use cache are incompatible"
+        if unused:
+            raise TypeError(f"Ignoring kwargs {unused}")
+        # check attention mask and invert
+        use_cache = cast_bool_to_primitive(use_cache)
+        if encoder_padding_mask is not None:
+            encoder_padding_mask = invert_mask(encoder_padding_mask)
+
+        # embed positions
+        positions = self.embed_positions(input_ids, use_cache=use_cache)
+
+        if use_cache:
+            input_ids = input_ids[:, -1:]
```

Comment:
I would actually advocated to delete these lines and try to make BART work with `decoder_input_ids.shape[-1] > 1` + `use_cache = True`. This is an edge case, but we had issues about this feature for other models, see: https://github.com/huggingface/transformers/issues/4368#issuecomment-630244541

Reply:
I'm not going to do that in this PR. It's already very large.  But I am down to cover the edge case follow up work for torch and TF at the same time. Made issue so I don't lose track: https://github.com/huggingface/transformers/issues/7814